docs: Extend documentation on prCreation (#19345)
renovatebot/renovate: docs/usage/configuration-options.md

Renovate defaults to `immediate` but you might want to change this to `not-pending` instead.

With prCreation set to `immediate`, you'll get a Pull Request and possible associated notification right away when a new update is available.
You'll have to wait until the checks have been performed, before you can decide if you want to merge the PR or not.

With prCreation set to `not-pending`, Renovate waits until tests pass or fail, and only then creates the PR.
When you get the PR notification, you can take action immediately, as you have the full test results.
If there are no checks associated, Renovate will create the PR once 24 hrs has elapsed since creation of the commit.

> With prCreation set to `status-success`, Renovate waits until all tests have finished running successfully, and only then creates the PR.

Reviewer: ```suggestion
With prCreation set to `status-success`, Renovate creates the PR only if/once all tests have passed.

For all cases of non-immediate PR creation, it does not mean that Renovate runs instantly once tests complete.
Instead, it means Renovate can create the PR on the next run after relevant tests have completed, so there will be some delay.
```

Renovate defaults to `immediate` but you might want to change this to `not-pending` instead.

With prCreation set to `immediate`, you'll get a Pull Request and possible associated notification right away when a new update is available.
You'll have to wait until the checks have been performed, before you can decide if you want to merge the PR or not.

> With prCreation set to `not-pending`, Renovate waits until all tests have finished running, and only then creates the PR.

Reviewer: ```suggestion
With prCreation set to `not-pending`, Renovate waits until tests pass or fail, and only then creates the PR.
```

When you get the PR notification, you can take action immediately, as you have the full test results.

> Particularly in conjunction with automerge settings, you're reducing the "noise" as PRs might not be created at all.

Reviewer: ```suggestion
```

This line is not correct. Branch automerge might prevent the PR from being created but it's got nothing to do with prCreating settings

Author: My consideration was, that with prCreation=not-pending and automerging, PRs would not get created at all. Compared to prCreation=immediate, this would actually reduce the "noise" of PRs getting created. Is my understanding wrong? "Reducing the noise" was already mentioned before my change - which was meant by that?

Otherwise I can simply remove the hint to automerge.

Reviewer: The first sentence here is also incorrect

> If there are no checks associated, Renovate will create the PR 24 hrs after the initial creation of the branch.

Reviewer: ```suggestion
If there are no checks associated, Renovate will create the PR once 24 hrs has elapsed since creation of the commit.
```
